### PR TITLE
[WIP] API

### PR DESCRIPTION
**Pull request by RJ-Clegg**
This WIP PR relates to [Issue 6](https://github.com/RocketFanOrg/RocketFanApp/issues/6)

Broken down as follows:

- [x] Add Framework 
- [ ] Add endpoint base 
- [ ] Add SpaceX Endpoints and map to our own endpoints 
- [ ] Networking layer
- [ ] Unit tests

### CODE REVIEW

File: API/API.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
+//
```
Comment (squarefrog) [inline API/API.h]: This can be removed

File: API/API.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+#import <UIKit/UIKit.h>
+
+//! Project version number for API.
+FOUNDATION_EXPORT double APIVersionNumber;
+
+//! Project version string for API.
+FOUNDATION_EXPORT const unsigned char APIVersionString[];
+
+// In this header, you should import all the public headers of your framework using statements like #import <API/PublicHeader.h>
```
Comment (squarefrog) [inline API/API.h]: Dont need this either

File: APITests/APITests.swift [hunk 3]
```diff
@@ -0,0 +1,34 @@
+//
```
Comment (squarefrog) [inline APITests/APITests.swift]: Delete the header

File: RocketFan.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -445,7 +633,9 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = CZG8CFZU5K;
```
Comment (squarefrog) [inline RocketFan.xcodeproj/project.pbxproj]: This can be removed I think

File: RocketFan.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -462,7 +652,9 @@
...
```
Comment (squarefrog) [inline RocketFan.xcodeproj/project.pbxproj]: Same here

File: RocketFan.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -558,6 +750,103 @@
...
 		};
+		E17553C72235324E0033C136 /* Debug */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				CLANG_ENABLE_MODULES = YES;
+				CODE_SIGN_IDENTITY = "";
+				CODE_SIGN_STYLE = Automatic;
+				CURRENT_PROJECT_VERSION = 1;
+				DEFINES_MODULE = YES;
+				DEVELOPMENT_TEAM = CZG8CFZU5K;
```
Comment (squarefrog) [inline RocketFan.xcodeproj/project.pbxproj]: Delete me

File: RocketFan.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -558,6 +750,103 @@
...
+				DYLIB_CURRENT_VERSION = 1;
+				DYLIB_INSTALL_NAME_BASE = "@rpath";
+				INFOPLIST_FILE = API/Info.plist;
+				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
+				LD_RUNPATH_SEARCH_PATHS = (
+					"$(inherited)",
+					"@executable_path/Frameworks",
+					"@loader_path/Frameworks",
+				);
+				PRODUCT_BUNDLE_IDENTIFIER = org.rocketfan.RocketFan.API;
```
Comment (squarefrog) [inline RocketFan.xcodeproj/project.pbxproj]: I wonder how to handle this. A user can specifiy their own bundle identifier. I'll have a think about this.

File: RocketFan.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -558,6 +750,103 @@
...
+				TARGETED_DEVICE_FAMILY = "1,2";
+				TEST_HOST = "$(BUILT_PRODUCTS_DIR)/RocketFan.app/RocketFan";
+			};
+			name = Debug;
+		};
+		E17553CC2235324E0033C136 /* Release */ = {
+			isa = XCBuildConfiguration;
+			buildSettings = {
+				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = CZG8CFZU5K;
```
Comment (squarefrog) [inline RocketFan.xcodeproj/project.pbxproj]: Another team line here

**Comment by RJ-Clegg** [comment]
Closing this as we made significant changes which won't make sense for this PR and the original branch it belonged to.

**Pull request closed without merging**